pennylane/registers.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module contains the :func:`registers` function to build the registers for a given dictionary of registers
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict, _start_wire_index=0):
+    """Returns the registers for the given dictionary of registers. The registers are a dictionary
+    of Wires objects where the key is the register name and the value is the Wires object. The
+    ordering of the Wires objects in the dictionary is based on appearance order first, then on
+    nestedness.
+
+    Args:
+        register_dict (dict[str, int]): dictionary of registers where the keys are the name of the
+        registers and the values are the number of qubits for said register
```

Comment:
I don't think we need to change the text too much as it's a complicated concept to convey, but here we're missing the ability for there to be nestedness, e.g., `{"alice": {"alice1": 10, "alice2:" 2}, "bob": 4, "cleo": 3}`.

Reply:
```suggestion
        register_dict (dict): a dictionary of registers where the keys are the names (str) of the
        registers and the values are either nested dictionaries of more registers or integers (int). At the most nested level for each register key, the value must be an int.
```

How about this?